packages/tests/tests/dns-peer-discovery.spec.ts

```diff
@@ -22,12 +23,13 @@ describe("DNS Discovery: Compliance Test", async function () {
   tests({
     async setup() {
       // create libp2p mock peerStore
-      const components = new Components({
+      const components = {
         peerStore: new PersistentPeerStore({
+          events: new EventEmitter(),
           peerId: await createSecp256k1PeerId(),
-          datastore: new MemoryDatastore(),
+          datastore: new MemoryDatastore() as any as Datastore,
```

Comment:
This is a strange thing. Thought `MemoryDatastore` is used in the `Libp2p` and correlates to proper interface - test env is complaining about wrong type so I used type casting for now.